### ISSUE TITLE
send_string and friends missing from type annotations

### ISSUE THREAD

**Issue by MechWarrior99** [zeromq/pyzmq#1643]
Hello, I installed pyzmq in to a virtual environment, and I am able to access the API's like `socket.send(b"Hellow World)` and that all works. 
However, I don't see other send methods like `.send_string(..)` and `.send_json(..)` that I see in the docs and in many examples.
I looked for all the documentation/resources online I could find but couldn't find anything about this. I am _assuming_ I missed something about how to go about installing it, but I simply cannot figure what it is.
I am admittedly still learning python, and also ZeroMQ, so I am not sure if this is even related to pyzmq specifically, and if that is the case I apologize now. I simply have no other ideas though.

Versions:
`libzmq-4.3.4`
`pyzmq-22.3.0`
`Python-3.9.9`

**Comment by MechWarrior99** [comment]
Yes? This is what I see. ( tried other editors and they all showed the same thing basically)
![4fyzZ4G](https://user-images.githubusercontent.com/8076495/146773323-feebf45f-ce28-4cc0-8c5e-327c909fc8f5.png)

It seems the code still executes without error? I was getting one earlier that it didn't know what `send_string()` was. But that was for a REP socket iirc?
```python
import zmq

context = zmq.Context()
socket = context.socket(zmq.REQ)
socket.connect("tcp://localhost:5555")

socket.send_string("World")
```

**Comment by minrk** [comment]
Ah, OK. The method's not missing, it might just be a problem in the new static analysis failing to _detect_ that the method is there. Type annotations are new in pyzmq, and something's probably missing somewhere.

**Comment by MechWarrior99** [comment]
Welp, that was it. When I got the error originally it must have been because I had a typo, or sent the wrong type of data or something. because of that I was sure that it simply wasn't there. I spent hours, and 3 different editors trying to figure out what I was doing wrong. Oh well, thanks for the help!

Would it be best to close this issue and open a new one specifically for the the Type annotation not covering the send methods?
